synapse/config/server.py

```diff
@@ -710,6 +710,11 @@ class LimitRemoteRoomsConfig:
             # Turn the list into a set to improve lookup speed.
             self.next_link_domain_whitelist = set(next_link_domain_whitelist)
 
+        templates_config = config.get("templates") or {}
+        self.custom_template_directory = templates_config.get(
+            "custom_template_directory"
+        )
```

Comment:
we should do some type-checking here: that `templates` is a `dict` and `custom_template_directory` is a string.